pkg/fleet/installer/installer.go

```diff
@@ -83,14 +84,50 @@ func NewInstaller(env *env.Env) (Installer, error) {
 	if err != nil {
 		return nil, fmt.Errorf("could not create packages db: %w", err)
 	}
-	return &installerImpl{
+	i := &installerImpl{
 		db:           db,
 		downloader:   oci.NewDownloader(env, http.DefaultClient),
 		repositories: repository.NewRepositories(paths.PackagesPath, paths.LocksPack),
 		configsDir:   paths.DefaultConfigsDir,
 		tmpDirPath:   paths.TmpDirPath,
 		packagesDir:  paths.PackagesPath,
-	}, nil
+	}
+	i.packageInstallers = map[string]packageInstaller{
+		packageDatadogAgent: &datadogAgentPackageInstaller{
+			basePackageInstaller: &basePackageInstaller{
+				installerImpl: i,
+				pkgName:       packageDatadogAgent,
+			},
+		},
+		packageDatadogInstaller: &datadogInstallerPackageInstaller{
+			basePackageInstaller: &basePackageInstaller{
+				installerImpl: i,
+				pkgName:       packageDatadogInstaller,
+			},
+		},
+		packageAPMInjector: &apmInjectorPackageInstaller{
+			basePackageInstaller: &basePackageInstaller{
+				installerImpl: i,
+				pkgName:       packageAPMInjector,
+			},
+		},
+	}
```

Comment:
I'm not convinced by this approach. I do like generic code but this is way too expensive to maintain (3 new files, new interfaces, a structure per package...). 

This is only for a single exception that will never happen again: for every package that is not the installer, the installer stays running and can control files & units independently. The only case where this doesn't happen is with the installer itself.

IMO we have to make it much simpler with something like
```
func stop_exp() {
  if installer {
    remove_symlink()
    stop_unit()
  } else {
    stop_unit()
    remove_symlink()
  }
}
```